asset-clustering/training-tuning-inference/inference.py

```
@@ -0,0 +1,83 @@
+import os
+import datetime
+import logging
+import numpy as np
+import cudf
+import cuml
+import cuml.preprocessing as cupreproc
+import pandas as pd
+import pickle
+import click
+import pdb
+from utils import compute_chars
+
+os.environ["CUDA_VISIBLE_DEVICES"] = '0'
+
+
+def normalize_host_data(data_fname_, norm_method='l2', preproc='minmax'):
+    df = cudf.read_csv(data_fname_)
+    print("Num. of columns:{}".format(len(df.columns)))
+
+    rm_assets = ['ActiveDirectory', 'EnterpriseAppServer']
+    print("\nREMOVED {} Assets from data".format(rm_assets))
+    df = df.loc[~df['LogHost'].isin(rm_assets)]
+
+    rm_cols = ['LogHost', 'uname_other_compacnt_login_frac', 'uname_that_compacnt_login_frac']
+    norm_cols_ = [x for x in df.columns if x not in rm_cols]
+
+    if preproc == 'unit_norm':
+        scaler = cupreproc.normalize(norm=norm_method)
+    elif preproc == 'minmax':
+        scaler = cupreproc.MinMaxScaler(feature_range=(0, 1))
+
+    df_norm = scaler.fit(df[norm_cols_]).transform(df[norm_cols_])
+    df_norm.columns = norm_cols_
+
+    return df, df_norm
+
+
+@click.command()
+@click.option('--model', default='dbscan', help='Clustering method to use.'\
+     ' Valid choices are \'kmeans\' or \'dbscan\'. Default is \'dbscan\'')
+def run(**kwargs):
+    model = kwargs['model']
+    compute_cluster_chars = True
+    assert model in ['kmeans', 'dbscan']
+
+    global NUM_DAYS, norm_cols
+    data_fname = '../datasets/host_agg_data_day-11_day-15.csv'
```

We can consider making this path a click option too (defaulting to the hard-coded value here) so users can test their own dataset easier.

---

Done.